bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/handler/PushbulletHandler.java

```diff
@@ -0,0 +1,218 @@
+        // check reconnect channel of the unifi binding for that
+
+        logger.debug("The Pushbullet binding is a read-only binding and cannot handle command '{}'.", command);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(PushbulletConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Name and Token are both "required", so set the Thing immediately ONLINE.
+        updateStatus(ThingStatus.ONLINE);
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(PushbulletActions.class);
+    }
+
+    public boolean sendPush(@Nullable String recipient, @Nullable String message, String type) {
+        return sendPush(recipient, "", message, type);
+    }
+
+    public boolean sendPush(@Nullable String recipient, @Nullable String title, @Nullable String message, String type) {
+        boolean result = false;
+
+        logger.debug("sendPush is called for ");
+        logger.debug("Thing {}", thing);
+        logger.debug("Thing Label: '{}'", thing.getLabel());
+
+        PushbulletConfiguration configuration = getConfigAs(PushbulletConfiguration.class);
+        logger.debug("CFG {}", configuration);
+        logger.debug("CFG Name  '{}'", configuration.getName());
+
+        try {
+            logger.debug("Recipient is '{}'", recipient);
+            logger.debug("Title is     '{}'", title);
+            logger.debug("Message is   '{}'", message);
+
+            Push push = new Push();
+            push.setTitle(title);
+            push.setBody(message);
+            push.setType(type);
+
+            if (recipient != null) {
+                if (isValidEmail(recipient)) {
+                    logger.debug("Recipient is an email address");
+                    push.setEmail(recipient);
+                } else if (isValidChannel(recipient)) {
+                    logger.debug("Recipient is a channel tag");
+                    push.setChannel(recipient);
+                } else {
+                    logger.warn("Invalid recipient: {}", recipient);
+                    logger.warn("Message will be broadcast to all user's devices.");
+                }
+            }
+
+            logger.debug("Push: {}", push);
+
+            String request = gson.toJson(push);
+            logger.debug("Packed Request: {}", request);
+
+            Properties headers = new Properties();
+            headers.put(HttpHeader.USER_AGENT, "openHAB / Pushbullet binding " + VERSION.toString());
```

Comment:
I suppose this toString would not be needed

Reply:
True, removed.